File: src/main/java/br/com/rotaractsorocabaleste/rotasales/event/EventController.java
package br.com.rotaractsorocabaleste.rotasales.event;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;
import java.util.Set;

@RestController
@RequestMapping("/events")
@RequiredArgsConstructor
public class EventController {

    private final EventService eventService;

    @GetMapping
    public Set<Event> findMyInstitutionActiveEvents() {
        return eventService.findMyInstitutionActiveEvents();
    }

    @PostMapping
    public Event create(@RequestBody @Valid final Event event) {
        return eventService.create(event);
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/entrypoint/controller/EventController.kt
package br.com.rotaractsorocabaleste.rotasales.entrypoint.controller

import br.com.rotaractsorocabaleste.rotasales.core.entity.Event
import br.com.rotaractsorocabaleste.rotasales.core.service.EventService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/events")
class EventController(
    private val eventService: EventService,
    private val logger: Logger = LoggerFactory.getLogger(EventController::class.java)
) {

    @GetMapping
    fun getActiveEvents(): ResponseEntity<List<Event>> {
        val ret = eventService.getActiveEvents()

        logger.info("Getting all active events, number of events=${ret.size}")

        return ResponseEntity.ok(ret)
    }

}